src/with-unique.js

```diff
@@ -1,14 +1,56 @@
-import { HTMLElement, sym, uniqueId } from './util';
+import { customElements, dashCase, HTMLElement, keys, sym } from './util';
+import { define } from './define';
 
 const _is = sym();
+const baseFunctionStatics = keys(function () {});
 
-export function withUnique (Base = HTMLElement) {
-  return class extends Base {
-    static get is () {
-      return this[_is] || (this[_is] = `x-${uniqueId().substring(0, 8)}`);
+let suffix = 0;
+
+function extendStatics (Ctor, Base) {
+  keys(Base).forEach(name => {
+    if (baseFunctionStatics.indexOf(name) > -1) {
+      return;
+    }
+    const descriptor = Object.getOwnPropertyDescriptor(Base, name);
+    if (descriptor) {
+      Object.defineProperty(Ctor, name, descriptor);
     }
-    static set is (is) {
-      this[_is] = is;
+  });
+}
+
+function formatName (prefix, suffix) {
+  return (
+    prefix.indexOf('-') === -1 ? `x-${prefix}` : prefix
+  ) + (
+    suffix ? `-${suffix}` : ''
+  );
+}
+
+function generateName (Ctor, hint) {
+  const prefix = hint || dashCase(Ctor.name) || 'element';
+  while (customElements.get(formatName(prefix, suffix))) {
+    suffix++;
+  }
+  return formatName(prefix, suffix++);
+}
+
+export function withUnique (Base = HTMLElement) {
+  function Ctor () {
+    const { constructor } = this;
+    define(constructor);
```

Comment:
this could be an issue for us at `blaze-elements` as we register classes to WC container only in root exports, otherwise we are using plain classes to be able to redefine them and not pollute customElements namespace registry

*example:*

**Button**

https://github.com/wc-catalogue/blaze-elements/blob/master/packages/button/index.ts#L5

-> user can:
`import {Button} from '@blaze-elements/button'`

which will give him registered WC element so he can use `<bl-button>click me </bl-button>`

but if he want's to register it via custom name, he can do:
`import Button from '@blaze-elements/button/Button'` , which will give him raw class, so he needs to do `customElements.define('my-button',Button)` -> `<my-button>I own you</my-button>`

this technique is further applied across other packages that are using existing WC classes like here 
https://github.com/wc-catalogue/blaze-elements/blob/master/packages/calendar/components/Button.tsx
-> this makes every package really atomic and self contained